File: 348-insertionSorting.cpp
#include<vector>
#include<iostream>

using DT=std::vector<int>;


void insertionSorting(DT& nums)  //insert the largest/smallest value into the selected section
{
    if(nums.empty())
        return ;
    
    int sz=nums.size();
    for(int i=1;i<sz;++i)
    {
        int minIdx=i-1,key=nums[i];  //we should save the nums[i] as one copy for following comparison, since it will be changed later
    
        while(minIdx>=0&&nums[minIdx]>=key)  //minIdx>=0&&nums[minIdx]>=nums[i] is wrong,should be minIdx>=0&&nums[minIdx]>=key
        {
            nums[minIdx+1]=nums[minIdx];
            minIdx--;
        }
        nums[minIdx+1]=key;
    }

}

int main()
{
    DT nums{3,5,2,9,1,6};
    insertionSorting(nums);
    for(auto i:nums)
        std::cout<<i<<',';
    return 0;
}

File: 348-insertionSorting.py
#simple insertion sorting algorithm

def insertionSorting(A):
    sz=len(A)
    if sz==0:
        return A
    for i in range(1,sz):
        key=A[i]
        j=i-1
        while j>=0 and key<A[j]:
            A[j+1]=A[j]
            j-=1
        
        A[j+1]=key  #j is -1 when is the smallest value
        
    return A


nums=[3,5,2,9,1,6];
newVals=insertionSorting(nums)
print(newVals)
